resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -133,7 +133,7 @@ private void query(InetSocketAddress nameServerAddr, final DnsQuestion question)
             public void operationComplete(Future<AddressedEnvelope<DnsResponse, InetSocketAddress>> future) {
                 queriesInProgress.remove(future);
 
-                if (promise.isDone()) {
+                if (promise.isDone() || future.isCancelled()) {
                     return;
```

Comment:
huh ? isDone() should be true if isCancelled() is be true. What I'm missing here ?